Documentation/coding-guidelines/adding-api-guidelines.md

```diff
@@ -0,0 +1,124 @@
+- If the library is not part of netstandard and not building against netstandard
+  - All the rules for a library that is not part of netstandard apply but instead of having a target framework of
+    `netstandard<latest>` it will have a target framework of `<framework><latest>`. Example `net<latest>` for desktop or `netcoreapp<latest>` for .NET Core.
+- It is common for a library to target multipe frameworks all the latest versions when adding new APIs.
+
+###Determine library version
+- If targeting netstandard
+  - Ensure minor version of the library is bumped since last stable package release
+- If targeting netcoreapp
+  - No version bump necessary
+
+##Making the changes in repo
+
+**If changing the library version**
+  - Update the `AssemblyVersion` property in `<Library>\dir.props` (ex: [System.Runtime\dir.props](https://github.com/dotnet/corefx/blob/master/src/System.Runtime/dir.props#L4)). If `AssemblyVersion` is
+    set in any of the individual csproj's remove it from them.
+
+**Update ref**
+  - If changing target framework
+    - `NugetTargetMoniker` in `<Library>\ref\<Library>.csproj` (ex: [ref\System.Runtime.csproj](https://github.com/dotnet/corefx/blob/master/src/System.Runtime/ref/System.Runtime.csproj#L8))
+    - Update framework section in `<Library>\ref\project.json` (ex: [System.Runtime\ref\project.json](https://github.com/dotnet/corefx/blob/master/src/System.Runtime/ref/project.json#L4))
+  - Add your API to `<Library>\ref\<Library>.cs`. Use GenAPI to normalize the file.
+  - If adding an additional target framework for the ref (ex: adding a .NET Core specific API) you may need
+  to add a `<Library>\ref\<Library>.builds` for the new configuration (ex: [ref/System.Runtime.Extensions.builds](https://github.com/dotnet/corefx/blob/master/src/System.Runtime.Extensions/ref/System.Runtime.Extensions.builds#L6)).
+
+**Update src**
+  - If changing target framework
+    - Update `NugetTargetMoniker` in `<Library>\src\<Library>.csproj` (ex: [src\System.Runtime.csproj](https://github.com/dotnet/corefx/blob/master/src/System.Runtime/src/System.Runtime.csproj#L12))
+    - Update framework section in `<Library>\src\project.json` (ex: [System.Runtime\src\project.json](https://github.com/dotnet/corefx/blob/master/src/System.Runtime/src/project.json#L3))
+    - Update build configurations in `<Library>\src\<Library>.builds` if necessary (see [Project configuration conventions](https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/project-guidelines.md#project-configuration-conventions))
+      - The default build configuration(i.e. TargetGroup excluded) should build the latest version
+      - If you need a live build of the older versions they need to be added as a new configuration
+        - Typically only done for libraries not in netstandard that have larger amounts of code
+          that we want to update outside of servicing events.
+  - Make your code changes.
```

Comment:
Should these source changes always be conditioned on $(TargetGroup) ?
  `<Compile Condition="'$(TargetGroup)'=='netstandard1.7'" Include="mynewapi.cs" />`